apps/system/js/sync_manager.js

```diff
@@ -727,36 +731,53 @@
       });
     },
 
-    saveDefaultSettings() {
-      var saveDefaultSetting = setting => {
+    saveSettings(prefix) {
+      var saveSetting = setting => {
         return new Promise(resolve => {
-          asyncStorage.getItem(setting, value => {
+          asyncStorage.getItem(prefix + setting, value => {
             if (value !== null) {
-              this.debug('Default setting value already saved', setting);
+              this.debug('Setting value already saved', setting);
               resolve();
               return;
             }
-            this.debug('Saving default setting value', setting);
-            asyncStorage.setItem(setting, this._settings[setting], resolve);
+            this.debug('Saving setting', prefix + setting,
+                       this._settings[setting]);
+            asyncStorage.setItem(prefix + setting, this._settings[setting],
+                                 resolve);
           });
         });
       };
 
       var promises = [];
       SyncManager.SETTINGS.forEach(setting => {
-        promises.push(saveDefaultSetting(setting));
+        promises.push(saveSetting(setting));
       });
 
       return Promise.all(promises);
     },
 
-    restoreDefaultSettings() {
+    saveDefaultSettings() {
+      this.debug('Saving default settings');
+      return this.saveSettings('');
+    },
+
+    saveUserSettings() {
+      this.debug('Saving user settings');
+      return this.saveSettings(this.user + '.');
+    },
+
+    restoreSettings(prefix) {
       var revertSetting = setting => {
         return new Promise((resolve, reject) => {
-          asyncStorage.getItem(setting, value => {
+          asyncStorage.getItem(prefix + setting, value => {
+            this.debug('Restoring setting ' + prefix + setting, value);
+            if (value === null) {
+              resolve();
+              return;
+            }
             // We need to remove the setting from asyncStorage to allow
             // default setting changes within OTA updates.
```

Comment:
I mean, do we want to reset each user's settings if we update the defaults through OTA? (maybe we do, so that we can add collections and switch them to read/write)